pkg/ovs/openflow/ofctrl_action.go

```diff
@@ -272,25 +293,38 @@ func (a *ofFlowAction) SetVLAN(vlanID uint16) FlowBuilder {
 
 // LoadARPOperation is an action to load data to NXM_OF_ARP_OP field.
 func (a *ofFlowAction) LoadARPOperation(value uint16) FlowBuilder {
-	loadAct, _ := ofctrl.NewNXLoadAction(NxmFieldARPOp, uint64(value), openflow13.NewNXRange(0, 15))
+	loadAct := &ofctrl.SetARPOpAction{Value: value}
 	a.builder.ApplyAction(loadAct)
 	return a.builder
 }
 
 // LoadRange is an action to load data to the target field at specified range.
 func (a *ofFlowAction) LoadRange(name string, value uint64, rng *Range) FlowBuilder {
-	loadAct, _ := ofctrl.NewNXLoadAction(name, value, rng.ToNXRange())
-	if a.builder.ofFlow.Table != nil && a.builder.ofFlow.Table.Switch != nil {
-		loadAct.ResetFieldLength(a.builder.ofFlow.Table.Switch)
+	hasMask := rng != nil
+	field, _ := openflow15.FindFieldHeaderByName(name, hasMask)
+	var valueBytes, maskBytes []byte
+	valueData := value
+	if hasMask {
+		mask := ^uint64(0) >> (64 - rng.Length()) << rng.Offset()
+		binary.BigEndian.PutUint64(maskBytes, mask)
+		field.Mask = util.NewBuffer(maskBytes)
+		valueData = valueData << rng.Offset()
 	}
-	a.builder.ApplyAction(loadAct)
-	return a.builder
+	binary.BigEndian.PutUint64(valueBytes, valueData)
```

Comment:
The program would panic if it ever uses this method, so apparently it's not using it. We should add unit tests for this package later to avoid such issues.

Reply:
Updated. I will add unit tests in a future patch.